### PR TITLE
Experiment: CMake support for iOS

### PR DESCRIPTION
**Pull request by oskirby**
## Description
I am just banging some code together to see if it's possible to build the iOS app using `cmake`. It seems like Qt/CMake support for iOS is still in its infancy, so there might be some rough patches and bugs to iron out before this can be considered viable.

At present, this requires Qt version 6.3.2 or later, and builds roughly as follows:
 - `mkdir build-ios`
 - `~/Qt/6.3.2/ios/bin/qt-cmake . -B build-ios -GXcode`
 - `open build-ios/Mozilla\ VPN.xcodeproj`
 - Select `mozillavpn` for `Any iOS Device (arm64)` and build.

## Reference
JIRA issue [VPN-2188](https://mozilla-hub.atlassian.net/browse/VPN-2188)

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

**Review by mcleinman — COMMENTED** [review]
Looking good so far. Thanks for working towards getting it running on simulator. A couple documentation nits for the moment.

File: README.md
Comment (oskirby) [reply README.md]: For the Qt builds I got from the official installer, Xcode was the default generator, but I agree it's better to make this explicit.

**Review by bakulf — APPROVED** [review]

**Review by bakulf — CHANGES_REQUESTED** [review]
You have to change `ci_scripts/ci_post_clone.sh` too

**Comment by brizental** [comment]
With regards to 246ea4b30b06543b2593414f5fca9a5ac942a37e, vpnglean is now supported on Android so IIUC this needs to be less generic and only skip vpnglean for iOS and Wasm specifically.

**Comment by oskirby** [comment]
@bakulf I am not sure how to test my changes to `ci_post_clone.sh` so they are just my best guess for now. I suppose that will be a project for future me once this gets merged onto a release branch and Xcode cloud actually starts to consume these changes.

**Comment by oskirby** [comment]
@brizental I've cherry-picked your PR #4880 to address your comment the `vpnglean` library.

File: vpnglean/CMakeLists.txt
Comment (brizental) [inline vpnglean/CMakeLists.txt:68]: Heh I was just having issues with this... 💯

**Review by brizental — APPROVED** [review]
Works for me! (Literally, I tested and it works for me).

**Comment by brizental** [comment]
Ok, with regards to my last comment. Indeed it works for me the application runs and all. However the logs seem to think a crash happened. This is what the XCode logs show for me when running the app on the simulator:

```
QML debugging is enabled. Only use this in a safe environment.
2022-11-09 12:13:28.082793+0100 Mozilla VPN[83990:4261408] Metal GPU Frame Capture Enabled
2022-11-09 12:13:28.083076+0100 Mozilla VPN[83990:4261408] Metal API Validation Enabled
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIView layer]
PID: 83990, TID: 4261792, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000103b8d1ea _ZL14layerForWindowP7QWindow + 122
5   Mozilla VPN                         0x0000000103b8d099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
6   Mozilla VPN                         0x0000000101950f65 _ZN15QSGRenderThread13syncAndRenderEv + 645
7   Mozilla VPN                         0x000000010195297a _ZN15QSGRenderThread3runEv + 298
8   Mozilla VPN                         0x00000001013a48f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
9   Mozilla VPN                         0x00000001013a371d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
10  Mozilla VPN                         0x00000001013a36af _ZN14QThreadPrivate5startEPv + 95
11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
2022-11-09 12:13:29.023622+0100 Mozilla VPN[83990:4261792] [reports] Main Thread Checker: UI API called on a background thread: -[UIView layer]
PID: 83990, TID: 4261792, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000103b8d1ea _ZL14layerForWindowP7QWindow + 122
5   Mozilla VPN                         0x0000000103b8d099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
6   Mozilla VPN                         0x0000000101950f65 _ZN15QSGRenderThread13syncAndRenderEv + 645
7   Mozilla VPN                         0x000000010195297a _ZN15QSGRenderThread3runEv + 298
8   Mozilla VPN                         0x00000001013a48f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
9   Mozilla VPN                         0x00000001013a371d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
10  Mozilla VPN                         0x00000001013a36af _ZN14QThreadPrivate5startEPv + 95
11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
PID: 83990, TID: 4261792, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000102f7d9d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
5   Mozilla VPN                         0x00000001036e4ede _ZNK7QWindow16devicePixelRatioEv + 94
6   Mozilla VPN                         0x0000000103b8db78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
7   Mozilla VPN                         0x0000000101951096 _ZN15QSGRenderThread13syncAndRenderEv + 950
8   Mozilla VPN                         0x000000010195297a _ZN15QSGRenderThread3runEv + 298
9   Mozilla VPN                         0x00000001013a48f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
10  Mozilla VPN                         0x00000001013a371d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
11  Mozilla VPN                         0x00000001013a36af _ZN14QThreadPrivate5startEPv + 95
12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
2022-11-09 12:13:29.145314+0100 Mozilla VPN[83990:4261792] [reports] Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
PID: 83990, TID: 4261792, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000102f7d9d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
5   Mozilla VPN                         0x00000001036e4ede _ZNK7QWindow16devicePixelRatioEv + 94
6   Mozilla VPN                         0x0000000103b8db78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
7   Mozilla VPN                         0x0000000101951096 _ZN15QSGRenderThread13syncAndRenderEv + 950
8   Mozilla VPN                         0x000000010195297a _ZN15QSGRenderThread3runEv + 298
9   Mozilla VPN                         0x00000001013a48f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
10  Mozilla VPN                         0x00000001013a371d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
11  Mozilla VPN                         0x00000001013a36af _ZN14QThreadPrivate5startEPv + 95
12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
```

The fun thing is, the app did not crash. It's running no problem. I can see no logs though. Looks related to this because it doesn't happen for qmake.

**Comment by brizental** [comment]
Ok with regards to my last comment, not sure this is related to this Pr or not, but I am not getting the same thing on qmake so here it goes. When I run the app it runs no issues... However the logs I see on XCode look like this:
```
QML debugging is enabled. Only use this in a safe environment.
2022-11-09 12:20:33.817438+0100 Mozilla VPN[93438:4284407] Metal GPU Frame Capture Enabled
2022-11-09 12:20:33.817696+0100 Mozilla VPN[93438:4284407] Metal API Validation Enabled
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIView layer]
PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000103e981ea _ZL14layerForWindowP7QWindow + 122
5   Mozilla VPN                         0x0000000103e98099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
6   Mozilla VPN                         0x0000000101c5bf65 _ZN15QSGRenderThread13syncAndRenderEv + 645
7   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
8   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
9   Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
10  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
2022-11-09 12:20:34.427591+0100 Mozilla VPN[93438:4284652] [reports] Main Thread Checker: UI API called on a background thread: -[UIView layer]
PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x0000000103e981ea _ZL14layerForWindowP7QWindow + 122
5   Mozilla VPN                         0x0000000103e98099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
6   Mozilla VPN                         0x0000000101c5bf65 _ZN15QSGRenderThread13syncAndRenderEv + 645
7   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
8   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
9   Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
10  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x00000001032889d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
5   Mozilla VPN                         0x00000001039efede _ZNK7QWindow16devicePixelRatioEv + 94
6   Mozilla VPN                         0x0000000103e98b78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
7   Mozilla VPN                         0x0000000101c5c096 _ZN15QSGRenderThread13syncAndRenderEv + 950
8   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
9   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
10  Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
11  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
2022-11-09 12:20:34.505929+0100 Mozilla VPN[93438:4284652] [reports] Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
Backtrace:
4   Mozilla VPN                         0x00000001032889d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
5   Mozilla VPN                         0x00000001039efede _ZNK7QWindow16devicePixelRatioEv + 94
6   Mozilla VPN                         0x0000000103e98b78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
7   Mozilla VPN                         0x0000000101c5c096 _ZN15QSGRenderThread13syncAndRenderEv + 950
8   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
9   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
10  Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
11  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
```
That is literally all, the actual VPN logs are supressed.

File: README.md
Comment (badboy) [inline README.md:360]: (drive-by comment). 
```suggestion
[Xcode](https://developer.apple.com/xcode/) version 12 or higher and [Qt](https://www.qt.io/download)
```

(I am going to die on this hill...)

**Review by bakulf — APPROVED** [review]

**Comment by mcleinman** [comment]
> Ok with regards to my last comment, not sure this is related to this PR or not, but I am not getting the same thing on qmake so here it goes. When I run the app it runs no issues... However the logs I see on XCode look like this:
> 
> ```
> QML debugging is enabled. Only use this in a safe environment.
> 2022-11-09 12:20:33.817438+0100 Mozilla VPN[93438:4284407] Metal GPU Frame Capture Enabled
> 2022-11-09 12:20:33.817696+0100 Mozilla VPN[93438:4284407] Metal API Validation Enabled
> =================================================================
> Main Thread Checker: UI API called on a background thread: -[UIView layer]
> PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
> Backtrace:
> 4   Mozilla VPN                         0x0000000103e981ea _ZL14layerForWindowP7QWindow + 122
> 5   Mozilla VPN                         0x0000000103e98099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
> 6   Mozilla VPN                         0x0000000101c5bf65 _ZN15QSGRenderThread13syncAndRenderEv + 645
> 7   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
> 8   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
> 9   Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
> 10  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
> 11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
> 12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
> 2022-11-09 12:20:34.427591+0100 Mozilla VPN[93438:4284652] [reports] Main Thread Checker: UI API called on a background thread: -[UIView layer]
> PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
> Backtrace:
> 4   Mozilla VPN                         0x0000000103e981ea _ZL14layerForWindowP7QWindow + 122
> 5   Mozilla VPN                         0x0000000103e98099 _ZN15QMetalSwapChain16surfacePixelSizeEv + 105
> 6   Mozilla VPN                         0x0000000101c5bf65 _ZN15QSGRenderThread13syncAndRenderEv + 645
> 7   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
> 8   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
> 9   Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
> 10  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
> 11  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
> 12  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
> =================================================================
> Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
> PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
> Backtrace:
> 4   Mozilla VPN                         0x00000001032889d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
> 5   Mozilla VPN                         0x00000001039efede _ZNK7QWindow16devicePixelRatioEv + 94
> 6   Mozilla VPN                         0x0000000103e98b78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
> 7   Mozilla VPN                         0x0000000101c5c096 _ZN15QSGRenderThread13syncAndRenderEv + 950
> 8   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
> 9   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
> 10  Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
> 11  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
> 12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
> 13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
> 2022-11-09 12:20:34.505929+0100 Mozilla VPN[93438:4284652] [reports] Main Thread Checker: UI API called on a background thread: -[UIView contentScaleFactor]
> PID: 93438, TID: 4284652, Thread name: QSGRenderThread, Queue name: com.apple.root.default-qos.overcommit, QoS: 0
> Backtrace:
> 4   Mozilla VPN                         0x00000001032889d1 _ZNK10QIOSWindow16devicePixelRatioEv + 33
> 5   Mozilla VPN                         0x00000001039efede _ZNK7QWindow16devicePixelRatioEv + 94
> 6   Mozilla VPN                         0x0000000103e98b78 _ZN15QMetalSwapChain14createOrResizeEv + 2120
> 7   Mozilla VPN                         0x0000000101c5c096 _ZN15QSGRenderThread13syncAndRenderEv + 950
> 8   Mozilla VPN                         0x0000000101c5d97a _ZN15QSGRenderThread3runEv + 298
> 9   Mozilla VPN                         0x00000001016af8f3 _ZZN14QThreadPrivate5startEPvENK3$_0clEv + 755
> 10  Mozilla VPN                         0x00000001016ae71d _ZN12_GLOBAL__N_122terminate_on_exceptionIZN14QThreadPrivate5startEPvE3$_0EEvOT_ + 29
> 11  Mozilla VPN                         0x00000001016ae6af _ZN14QThreadPrivate5startEPv + 95
> 12  libsystem_pthread.dylib             0x00007fff701cb4e1 _pthread_start + 125
> 13  libsystem_pthread.dylib             0x00007fff701c6f6b thread_start + 15
> ```
> 
> That is literally all, the actual VPN logs are supressed.

@brizental I've run into this as well. I believe that we only show logs in the Xcode console if you're hitting a staging server. Alternatively, in `loghandler.cpp`, change `if (!Constants::inProduction())` to `if (true)`.

**Comment by mcleinman** [comment]
I'm currently getting this error when trying to build:
```
CMake Error at vpnglean/CMakeLists.txt:51 (message):
  Unable to determine Rust target for iOS ().
```

**Comment by mcleinman** [comment]
> I'm currently getting this error when trying to build:
> 
> ```
> CMake Error at vpnglean/CMakeLists.txt:51 (message):
>   Unable to determine Rust target for iOS ().
> ```
> 
> (Re-creating the symlink doesn't fix it.)

I'm still getting this same error - anything I should try on my end?

**Comment by oskirby** [comment]
@mcleinman the latest commit 00ad8905c199f7fa66b027ff359f4de70b944138 should get this working again on M1 macbooks.

**Comment by oskirby** [comment]
@brizental the default build, unless otherwise specified generates debug information (eg: `MVPN_DEBUG` is defined) and will be configured to connect to Guardian production. I think in the case of the iOS simulator there is a reasonable argument to be made that we should target staging by default.

I also have a pet peeve that we silence the console output on all builds, I would prefer the client to be noisy when either built for debug or when connecting to staging.

File: README.md
Comment (mcleinman) [inline README.md:357]:
```suggestion
We also support building from sources for iOS using CMake.
```

Isn't the WIP and "please update if things change" true of all our documentation?

File: README.md
Comment (mcleinman) [inline README.md:361]:
```suggestion
1. Prepare: Ensure
[Xcode](https://developer.apple.com/xcode/) version 12 or higher and [Qt](https://www.qt.io/download)
version 6.3.2 are both installed.

Add the rust target for iOS devices:
`rustup target add x86_64-apple-ios && rustup target add aarch64-apple-ios`
```

File: README.md
Comment (mcleinman) [inline README.md:366]: The GH suggestion syntax conflicts with the readme bash syntax, hence the dashes thrown in here - will need to remove.

```suggestion
2. Use `qt-cmake` (from the Qt installation) to build and configure the Xcode project:
```bash
/Users/example/Qt/6.3.2/ios/bin/qt-cmake . -B build-ios -GXcode
-```

This builds for an iOS device. Due to cmake limitations, we need to include a build variable for building for the simulator. To make it easier to develop, it's recommended to build to a different folder/project for simulator, so that switching between device and simulator testing does not require re-building the project:
```bash
/Users/example/Qt/6.3.2/ios/bin/qt-cmake . -B build-ios-sim -GXcode CMAKE_OSX_SYSROOT="iphonesimulator"
-```
```

File: README.md
Comment (mcleinman) [inline README.md:377]:
```suggestion
3. Open the generated Xcode project with `open build-ios/Mozilla\ VPN.xcodeproj`. (Or `open build-ios-sim/Mozilla\ VPN.xcodeproj` for the suggested simulator-specific directory.)
```

File: README.md
Comment (mcleinman) [inline README.md:379]: Do we want to recommend the `Any iOS Device` setting? For building for testing, don't we want the specific device/simulator?

**Review by mcleinman — COMMENTED** [review]
Suggestions for readme

**Review by mcleinman — COMMENTED** [review]
Looking good to me. After the readme changes are made (mentioned in a separate review), I'm ready to approve. 

(I definitely want a second approver on this - it works for me, and seems to look good, but I'm new to cmake/qmake and so more eyes are good.)

File: README.md
Comment (oskirby) [reply README.md:379]: I think the `Any iOS Device` setting is how Xcode tries to build for real devices, and we are trying to make sure that this setting matches what cmake is expecting at configuration time via the `CMAKE_OSX_SYSROOT` option. Unwinding the hardcoded nature of this sysroot setting is going to require some work on supporting multiple-architecture builds.

**Review by mcleinman — APPROVED** [review]
With those readme changes, I'm giving this my approval. Thanks for this Owen!

Still would love another approver on this before we merge.

**Comment by brizental** [comment]
> Still would love another approver on this before we merge.

Not sure about @bakulf 's approval, but mine still stands FWIW :)

**Pull request merged**
